plugins/rocksdb/src/main/java/org/hyperledger/besu/plugin/services/storage/rocksdb/segmented/RocksDBColumnarKeyValueStorage.java

```diff
@@ -258,10 +256,13 @@ public void remove(final ColumnFamilyHandle segment, final byte[] key) {
     }
 
     @Override
-    public void commit() throws StorageException {
+    public void commit() {
       try (final OperationTimer.TimingContext ignored = metrics.getCommitLatency().startTimer()) {
         innerTx.commit();
       } catch (final RocksDBException e) {
+        if (e.getMessage().contains("No space left on device")) {
```

Comment:
should we add this check on put, remove, and rollback?  as well as `org.hyperledger.besu.plugin.services.storage.rocksdb.unsegmented.RocksDBTransaction`?